### PR TITLE
FE#49 forgot password

### PR DESCRIPTION
**Pull request by logicplace**
There are two possible modes for requesting the reset:
1. Enter nothing and click ?, a new form will appear where you can enter the username.
2. Enter the username and click ?, a password will be sent directly, no special form.

In both cases a message appears indicating success.

Once the email is received, click the link (or modify the host accordingly) and it will take you to the reset page. Enter the new password twice and hit reset.

Try logging in again afterwards to make sure. I changed welches's password this way so you're gonna have to!

There's no real time password sameness confirmation because I didn't have any graphics on hand and I figured it wasn't important enough right now.

### CODE REVIEW

File: js/index.js [hunk 1]
```diff
@@ -50,4 +50,48 @@ $(function() {
...
+            $("#loginError").text("Error requesting password.");
+        });
+    });
 });
+
+function revertToLogin() {
+    $(".resetItem").hide();
+    $(".loginItem").show();
+    $("#resetButton").prop("disabled", false);
+}
```
Comment (scotte216) [inline js/index.js]: Newline at EOF

File: js/forgot.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            "type": "POST",
+        }).done(function(json) {
+            localStorage.accessToken = json.token;
+            window.location.href = "index.html";
+        }).fail(function() {
+            // Show error message
+            $("#resetError").text("Error logging in.");
+        });
+    });
+});
```
Comment (scotte216) [inline js/forgot.js]: Newline at EOF

File: css/index.css [hunk 3]
```diff
@@ -122,3 +126,7 @@ input.button:active {
     border: darkolivegreen solid 2px;
     background: dimgray;
 }
+
+.resetItem {
+    display: none;
+}
```
Comment (scotte216) [inline css/index.css]: Newline at EOF

**Comment by scotte216** [comment]
Not sure if this is a problem but a valid username with extra white space (at the end) fails as invalid username. I mean, technically it IS :D Just thought I'd mention something I found in testing.

**Comment by scotte216** [comment]
Also, if you do something incorrectly (like add a white space) and get an invalid user, the message lingers where I don't expect. See pictures!

step 1

step 2

step 3

step 4

**Comment by scotte216** [comment]
Otherwise looks good to me!

**Comment by logicplace** [comment]
The space in username thing is also true for logging in, so I'm not sure if I want to handle it in this PR.

Your other suggestions have been accounted for now, though.

**Comment by scotte216** [comment]
My commit message is slightly erroneous. I didn't correct a typo in the message but in the object name.

🍷

**Comment by scotte216** [comment]
I have no idea what's going on... I tried to push the merge conflict we worked on, and now it magically says the PR was merged. I'm going to tap out before I make more of a mess!

**Pull request merged**